### DISCUSSION TITLE
Multiple Commit Selection doesnt support scroll

### DISCUSSION THREAD

**Discussion by Madalosso** [community/community#13028] [Pull Requests]
When reviewing a PR, on the Files Changed page, there's a menu that allows the reviewer to select the changes for

* All Changes
* Changes since your last review
* And we can select a specific commit or a range of commits.

The problem starts when I'm trying to select a range of commits, but I need to scroll the list of commits down to pick a specific one.
I can scroll down the list of commits while holding shift (which is the key that enables the multi-selection on commits), and if I release shift to try to press it again after scrolling, the feature won't work.

I don't know if I should list this as a discussion or a bug, but either way, this feature needs to support scrolling.
Tested on Chrome Version 98.0.4758.102 (Official Build) (64-bit)

**Comment by charlievrettos** [comment]
This seems to be happening to me even when I don't scroll. Selecting 6 of 15 commits, without scrolling, for example results in only 1 or 2 commits selected.

Tested on macOS Monterey 12.0.1
- Chrome Version 99.0.4844.51 (Official Build) (arm64)
- Safari Version 15.1 (17612.2.9.1.20)

**Comment by enricobattocchi** [comment]
The same is happening on Firefox 99.0 on Linux.

**Comment by bheston** [comment]
I'm experiencing the same as mentioned above.

Additionally, the interaction model is complex and should be reevaluated from an accessibility perspective. I can't figure out how to activate the feature with a keyboard and attempting to use it with a mouse is proving difficult or impossible.

Further, I really just want "Show changes since your last review", but that results in the dreaded "We went looking everywhere" message. I suspect that feature doesn't work on rebased branches as I frequently run into that with links from email notifications and my team rebases as a general rule.

**Comment by sezanzeb** [comment]
I went around this problem by inserting commit hashes into the url.

Making the list scroll down when moving the mouse to the bottom of the dropdown would also be cool while holding shift.

Or support selecting multiple commits by holding down shift and using arrow keys.

**Reply by Madalosso** [reply]
That's a nice alternative to get the functionality working. thanks for sharing.

**Reply by richfromm** [reply]
Thank you! Somewhat annoying, but great to have a workaround.

**Comment by jlevitt** [comment]
Also broken for me on Windows 11/Chrome.

**Comment by akgerber** [comment]
This remains broken for me, with or without scrolling.

**Comment by richfromm** [comment]
I've always found the multi-commit (but less than "all commits") selection within a PR to be hard to use, but indeed now I am finding it completely impossible if the range of commits requires scrolling.

macOS Sonoma 14.4.1, Chrome Version 124.0.6367.93 (Official Build) (arm64)

**Comment by tvsaru** [comment]
Came looking for a solution to this problem, but found this unresolved ticket.
On Sonoma 14.4.1 and Firefox 125.0.2.

**Comment by vanduyta** [comment]
Any update on this? Quite a basic functionality isn't it?

**Comment by adamreisnz** [comment]
This is so painful 😥

**Comment by Cody-G-G** [comment]
This is also completely blocking me from using the commit range selection feature whenever there are more than a couple commits, which is exactly when it would be most powerful....

EDIT: this method solved it for me as a workaround - https://stackoverflow.com/questions/29518780/way-to-compare-range-of-commits-in-pull-request-on-github#:~:text=GitHub%20has%20a%20built%2Din,shift%20)%20click%20the%20end%20commit.

**Comment by kyletolle** [comment]
This bug has frustrated me for quite a while. I have found, however, that I can Shift+Click and then scroll down the list if I use my MacBook Pro's trackpad. Scrolling using the mouse wheel does not work, but using the track pad's scroll gesture does. Perhaps it'd work for other devices with trackpads as well? This won't help everyone, since not everyone will have a trackpad to use, but I hope it'll help someone. The other workarounds here are helpful as well, until a real solution is released by GitHub.
